planet_scope/ndvi_statistics/script.js

```diff
@@ -0,0 +1,55 @@
+//VERSION=3
+//NDVI
+
+function setup() {
+  return {
+    input: [{
+      bands: [
+        "Red",
+        "NIR",
+        "dataMask", 
+        "UDM2_Clear"
+      ]
+    }],
+    output:  [
+      { id: "default", bands: 4 },
+      { id: "index", bands: 1, sampleType: "FLOAT32" },
+      { id: "eobrowserStats", bands: 2, sampleType: "FLOAT32" },
+      { id: "dataMask", bands: 1 },
+    ]
+  }
+}
+
+
+function isClear (udm2_clear) {
+  if (udm2_clear == 0)  {
+    return false;
+  }
+  else {
+    return true;
+  }
+}
+  
+function evaluatePixel(samples) {
+  
+    let ndvi = (samples.NIR-samples.Red)/(samples.NIR+samples.Red);
+
+    const indexVal = samples.dataMask === 1 ? ndvi : NaN;
+
+    const clear = isClear(samples.UDM2_Clear)?0:1
```

Comment:
Since `UDM2_Clear` is either 0 (not clear) or 1 (clear) this can be omitted completely. Instead the multiplication that is done 3 times below can be done once here.

```suggestion
const clear = samples.dataMask * samples.UDM2_Clear;
```